app/domain/etl/ga/user_feedback_processor.rb

```diff
@@ -45,7 +45,7 @@ def load_metrics_query(date_to_s)
       UPDATE facts_metrics
       SET useful_no = s.useful_no,
           useful_yes = s.useful_yes,
-          satisfaction= s.useful_yes / (s.useful_yes + s.useful_no::float)
+          satisfaction= s.useful_yes / NULLIF((s.useful_yes + s.useful_no::float),0)
```

Comment:
What will happen for pages where the actual value for the day is 0 interactions with the usefulness metrics? In my mind, they should be recorded as 0, not NULL, but I think this would record them as NULL?

Reply:
I'd argue that for days we have zero interactions, we have no data, if we have no data, the satisfaction score is unknown. Saying a page has 0% satisfaction based on no data would be misleading and inaccurate.

Reply:
Ok, yeah, I'm not really sure what was happening before though.

To avoid a division by 0 error, somehow `useful_yes` in `events_gas` must be 0, but `useful_no` must be > 0...

Reply:
@cbaines The NULLIF handles most cases, except `NULL\NULL` which I'll need to think a bit more about.

Reply:
If you want to be consistent with the aggregation views:
```sql
CASE
  WHEN useful_yes + useful_no = 0 THEN NULL
  ELSE useful_yes / (useful_yes + useful_no)::float
END
```

Reply:
NULL / NULL should be okay because of the type cast

Reply:
Ah my local testing omitted the casting. Awesome!